src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiClient.java

```diff
@@ -262,10 +263,13 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
     public List<BitbucketPullRequestValue> getPullRequests() throws InterruptedException, IOException {
         List<BitbucketPullRequestValue> pullRequests = new ArrayList<>();
 
+        // we can not use the default max pagelen also if documented
+        // https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/pullrequests
+        int pageLen = 50;
```

Comment:
The link above does not reference anything about page length 😕

Reply:
The link for the GET reports "pagelen:integer 1..∞
Current number of objects on the existing page. The default value is 10 with 100 being the maximum allowed value. Individual APIs may enforce different values."

But this is not true. If pagelen is greater than 50 I got am HTTP 400 for pagelen value. For this reason I had keep 50 as max value. Maybe wrong BB documention or atlassian forgot to increase the limit for this API.

Reply:
ctrl+f does not find any mention of "pagelen"

Reply:
This link is more correct: https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/pullrequests#get

Reply:
Tomorrow when I at the office I will change the comment and rebase to master

Reply:
no need to rebase, the PR will be squashed

Reply:
comment changed and rebased. FF merge is available.